third-party/realdds/src/dds-serialization.cpp

```diff
@@ -0,0 +1,245 @@
+{
+    os << "'" << info.topicName() << "' [";
+    os << /*"type " <<*/ info.typeName();
+    os << /*" reliability"*/ " " << info.m_qos.m_reliability;
+    if( ! ( info.m_qos.m_durability == eprosima::fastdds::dds::DurabilityQosPolicy() ) )
+        os << /*" durability"*/ " " << info.m_qos.m_durability;
+    os << "]";
+    return os;
+}
+
+
+std::ostream & operator<<( std::ostream & os, ReaderProxyData const & info )
+{
+    os << "'" << info.topicName() << "' [";
+    os << /*"type " <<*/ info.typeName();
+    os << /*" reliability"*/ " " << info.m_qos.m_reliability;
+    if( !(info.m_qos.m_durability == eprosima::fastdds::dds::DurabilityQosPolicy()) )
+        os << /*" durability"*/ " " << info.m_qos.m_durability;
+    os << "]";
+    return os;
+}
+
+
+}  // namespace rtps
+}  // namespace fastrtps
+}  // namespace eprosima
+
+
+namespace realdds {
+
+
+eprosima::fastdds::dds::ReliabilityQosPolicyKind reliability_kind_from_string( std::string const & s )
+{
+    if( s == "best-effort" )
+        return eprosima::fastdds::dds::BEST_EFFORT_RELIABILITY_QOS;
+    if( s == "reliable" )
+        return eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS;
+    DDS_THROW( runtime_error, "invalid reliability kind '" << s << "'" );
+}
+
+
+eprosima::fastdds::dds::DurabilityQosPolicyKind durability_kind_from_string( std::string const & s )
+{
+    if( s == "volatile" )
+        return eprosima::fastdds::dds::VOLATILE_DURABILITY_QOS;
+    if( s == "transient-local" )
+        return eprosima::fastdds::dds::TRANSIENT_LOCAL_DURABILITY_QOS;
+    if( s == "transient" )
+        return eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS;
+    if( s == "persistent" )
+        return eprosima::fastdds::dds::PERSISTENT_DURABILITY_QOS;
+    DDS_THROW( runtime_error, "invalid durability kind '" << s << "'" );
+}
+
+
+eprosima::fastdds::dds::HistoryQosPolicyKind history_kind_from_string( std::string const & s )
+{
+    if( s == "keep-last" )
+        return eprosima::fastdds::dds::KEEP_LAST_HISTORY_QOS;
+    if( s == "keep-all" )
+        return eprosima::fastdds::dds::KEEP_ALL_HISTORY_QOS;
+    DDS_THROW( runtime_error, "invalid history kind '" << s << "'" );
+}
+
+
+eprosima::fastrtps::rtps::MemoryManagementPolicy_t history_memory_policy_from_string( std::string const & s )
+{
+    if( s == "preallocated" )
+        return eprosima::fastrtps::rtps::PREALLOCATED_MEMORY_MODE;
+    if( s == "preallocated-with-realloc" )
+        return eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+    if( s == "dynamic-reserve" )
+        return eprosima::fastrtps::rtps::DYNAMIC_RESERVE_MEMORY_MODE;
+    if( s == "dynamic-reusable" )
+        return eprosima::fastrtps::rtps::DYNAMIC_REUSABLE_MEMORY_MODE;
+    DDS_THROW( runtime_error, "invalid history memory policy '" << s << "'" );
+}
+
+
+void reliability_qos_from_json( eprosima::fastdds::dds::ReliabilityQosPolicy & qos, nlohmann::json const & j )
```

Comment:
According to the name I expected an object to be created and returned.
Maybe `override_qos_from_json` or `override_based_on_json` with overloads for the different policies?

Reply:
I didn't want to use overloading.
Adding `override` will make the names longer... but OK, changed.